openwisp_users/tests/test_admin.py

```diff
@@ -1275,6 +1275,76 @@ def test_only_superuser_can_delete_inline_org_owner(self):
             self.assertEqual(r.status_code, 200)
             self.assertContains(r, '-DELETE">Delete')
 
+    def test_delete_org_user(self):
+        self.client.force_login(self._get_admin())
+        user1 = self._create_user(username='user1', email='user1@email.com')
+        org1 = self._create_org(name='org1')
+        org_user = self._create_org_user(user=user1, organization=org1, is_admin=True)
+
+        with self.subTest('test delete org user which belongs to owner'):
+            post_data = {'post': 'yes'}
+            url = reverse(
+                f'admin:{self.app_label}_organizationuser_delete', args=[org_user.pk]
+            )
+            r = self.client.post(url, post_data, follow=True)
+            qs = OrganizationUser.objects.filter(organization=org1, user=user1)
+            self.assertEqual(r.status_code, 200)
+            msg = (
+                't delete this organization user because it '
+                'belongs to an organization owner'
+            )
+            self.assertContains(r, msg)
+            self.assertEqual(qs.count(), 1)
+
+        with self.subTest('test delete org user which belongs to no owner'):
+            org2 = self._create_org(name='org2')
+            org_u = self._create_org_user(user=user1, organization=org2, is_admin=False)
+            post_data = {'post': 'yes'}
+            url = reverse(
+                f'admin:{self.app_label}_organizationuser_delete', args=[org_u.pk]
+            )
+            r = self.client.post(url, post_data, follow=True)
+            qs = OrganizationUser.objects.filter(organization=org2, user=user1)
+            self.assertEqual(r.status_code, 200)
+            self.assertContains(r, 'was deleted successfully.')
+            self.assertEqual(qs.count(), 0)
+
+        with self.subTest("Can not delete only owner's org user with action"):
+            post_data = {
+                'action': 'delete_selected_overridden',
+                '_selected_action': [org_user.pk],
+                'post': 'yes',
+            }
+            url = reverse(f'admin:{self.app_label}_organizationuser_changelist')
+            r = self.client.post(url, post_data, follow=True)
```

Comment:
This is the point which calls `delete_selected_overridden`. with the `delete_selected_overridden`, the number of queries this line shows is `8` and when `delete_selected_overridden` is commented out (that is using the default `delete_selected`) the number of queries returned is `23`.